### PR TITLE
zenroom: init at 3.5.0

### PR DESCRIPTION
**Pull request by jacksmithinsulander**
###### Description of changes

I have added the zenroom interpreter / execution environment for zencode, as a nix derivation. Zenroom is a neat smart contract execution plattform.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/zenroom/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "zenroom";
+  version = "3.5.0";
+
+  src = fetchurl {
+    url = "https://files.dyne.org/zenroom/nightly/zenroom-linux-amd64";
+    hash = "sha256-9xhQPrez4pk1zCzCyBqmNF/mkfChcA38p/s8//AwpZ8=";
+  };
```
Comment (Janik-Haag) [inline pkgs/development/interpreters/zenroom/default.nix:13]: Why aren't you fetching the release from github? https://github.com/dyne/Zenroom

Comment (jacksmithinsulander) [reply pkgs/development/interpreters/zenroom/default.nix:13]: I tried to use that at first, but I had a lot of issues with building it using cmake

Comment (jacksmithinsulander) [reply pkgs/development/interpreters/zenroom/default.nix:13]: Do I need to do something in order to get this merged?

File: pkgs/development/interpreters/zenroom/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      Zenroom is a tiny secure execution environment that integrates in any
+      platform and application, even on a chip or a web page.
+      It can authenticate, authorize access and execute human-readable
+      smart contracts for blockchains, databases and much more.
+      It helps to develop cryptography keeping it simple, understandable and
+      maintainable.
+    '';
+    license = licenses.agpl3Plus;
+    maintainers = with maintainers; [ jacksmithinsulander ];
+    mainProgram = finalAttrs.pname;
```
Comment (drupol) [inline pkgs/development/interpreters/zenroom/default.nix:35]: I don't think this is needed.

**Review by drupol — CHANGES_REQUESTED** [review]
Thanks for your contribution.

Could you please rebase and sort your commits so in the end there's only 2 commits in your PR.

1. One commit for adding yourself in the maintainers list
2. One commit for adding `zenroom`

**Pull request closed without merging**